doc/source/guide/compiling.rst

```diff
@@ -0,0 +1,159 @@
+.. _compiling:
+
+Compiling for Coverage
+======================
+
+In order to collect coverage data,
+your software must be “instrumented” by the compiler.
+That means, you must re-compile your software with special compiler options.
+
+The general workflow is:
+
+1. compile your software to enable coverage profiling
+2. execute your software to collect coverage profiles
+3. run ``gcovr`` to create reports from the collected coverage profiling data
+
+This document explains how you can use GCC or Clang
+to compile with coverage instrumentation.
+
+If you cannot compile your software with coverage flags,
+you cannot use gcovr.
+However, other tools like kcov_ might help.
+
+.. _kcov: https://github.com/SimonKagstrom/kcov
+
+
+Example Code
+------------
+
+The following ``example.cpp`` program
+is used to illustrate the compilation process:
+
+.. include:: ../../examples/example.cpp
+    :code: cpp
+    :number-lines: 1
+
+This code executes several subroutines in this program,
+but some lines in the program are not executed.
+
+
+Compiler Options
+----------------
+
+We compile ``example.cpp`` with the GCC compiler as follows:
+
+.. include:: ../../examples/example.sh
+    :code: bash
+    :start-after: #BEGIN compile
+    :end-before: #END compile
+
+What do these compiler flags mean?
+
+* We compile without optimization (``-O0``),
+  because optimizations may merge lines of code
+  or otherwise change the flow of execution in the program.
+  This can change the measured coverage.
+
+  If you are having problems with lots of uncovered branches,
+  see: :ref:`exception branches`
+
+* Either ``--coverage`` or  ``-fprofile-arcs -ftest-coverage`` are needed
+  so that the compiler produces the information necessary to gather coverage data.
+  The compiler will immediately create a ``example.gcno`` file with metadata.
```

Comment:
Should we mention that this file is created for each object?

Reply:
Very good point, I'll write a section about using multiple object files.

Reply:
Done! The guide page for compilation options was expanded with a section “Working with Multiple Object Files”.